### PR TITLE
Spring boot branch

### PR DESCRIPTION
**Pull request by KajalMathur**
Kindly review my product service and provide your review comments.

### CODE REVIEW

File: src/main/java/com/example/demo/controller/WebController.java [hunk 1]
```diff
@@ -12,63 +9,43 @@
...
-	      public ResponseEntity<Object> delete(@PathVariable("id") int id) { 
-	    	  log.debug("Inside HTTP delete method");
-	    	  ps.deleteProductById(id);
-	    	   return  new ResponseEntity<>("Product is Deleted successfully" ,HttpStatus.OK);
-	         // return new ResponseEntity<>(ps.productServiceDelete(id), HttpStatus.OK);
-	      }
+
+	@Autowired
+	// private ProductService // for the cache implementation
+	ProductServiceimpl ps;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: class name should be pascal case.. ProductServiceImpl, and please use sensible variable names too. And remove the commented line above

File: src/main/java/com/example/demo/controller/WebController.java [hunk 2]
```diff
@@ -12,63 +9,43 @@
...
+
+	// get
+	@RequestMapping(value = "/products")
+
+	public ResponseEntity<Object> getProduct() {
+
+		return new ResponseEntity<>(ps.getAllProducts(), HttpStatus.OK);
+	}
+
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: no blank space on line 24.
The method should not be returning an Object.. it should be exactly the type of ResponseEntity being returned

File: src/main/java/com/example/demo/controller/WebController.java [hunk 3]
```diff
@@ -12,63 +9,43 @@
...
-	    	   return  new ResponseEntity<>("Product is Deleted successfully" ,HttpStatus.OK);
-	         // return new ResponseEntity<>(ps.productServiceDelete(id), HttpStatus.OK);
-	      }
+
+	@Autowired
+	// private ProductService // for the cache implementation
+	ProductServiceimpl ps;
+
+	// get
+	@RequestMapping(value = "/products")
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: Use @GetMapping here like you used in your other project

File: src/main/java/com/example/demo/controller/WebController.java [hunk 4]
```diff
@@ -12,63 +9,43 @@
...
+
+	public ResponseEntity<Object> getProduct() {
+
+		return new ResponseEntity<>(ps.getAllProducts(), HttpStatus.OK);
+	}
+
+	// POST
+	@RequestMapping(value = "/products", method = RequestMethod.POST)
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: @PostMapping here

File: src/main/java/com/example/demo/controller/WebController.java [hunk 5]
```diff
@@ -12,63 +9,43 @@
...
+	public ResponseEntity<Object> createProduct(@RequestBody Product product) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: Dont return `ResponseEntity<Object>` when you can return the exact type `ResponseEntity<String>`

Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: Dont you need @Vaid before Product product ? Are the @NotNull @NotBlank annotations working fine ?

File: src/main/java/com/example/demo/controller/WebController.java [hunk 6]
```diff
@@ -12,63 +9,43 @@
...
+		ps.createProduct(product);
+		return new ResponseEntity<>("Product is created successfully", HttpStatus.CREATED);
+	}
+
+	// PUT
+	@RequestMapping(value = "/products/{id}", method = RequestMethod.PUT)
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: @PutMapping

File: src/main/java/com/example/demo/controller/WebController.java [hunk 7]
```diff
@@ -12,63 +9,43 @@
...
+	public String updateProduct(@PathVariable("id") int id, @RequestBody Product product) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: I think, if your variable name is EXACTLY same as the pathVariable in the `/products/{id}` then 
`@PathVariable("id") int id` is not needed, just `@PathVariable int id` would do

File: src/main/java/com/example/demo/controller/WebController.java [hunk 8]
```diff
@@ -12,63 +9,43 @@
...
+		ps.updateProduct(product);
+		return "Product is updated successfully";
+	}
+
+	// delete
+	@RequestMapping(value = "/products/{id}", method = RequestMethod.DELETE)
+	public ResponseEntity<Object> delete(@PathVariable("id") int id) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/controller/WebController.java]: same comments as above APIs

File: src/main/java/com/example/demo/model/Manufacturer.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+@Document
+public class Manufacturer {
+
+	@Id
+	int manufactureId;
+
+	@NonNull
+	String manufactureName;
+
+	@Range(min = 0l, message = "Please select positive numbers Only")
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/model/Manufacturer.java]: Niceeee.......

File: src/main/java/com/example/demo/model/Product.java [hunk 10]
```diff
@@ -1,72 +1,43 @@
...
-	@NonNull
-	@JsonProperty("papappa")
-	List <String> othersize;
-	
-	
-	
 
 	@NonNull
-	Map<String , String> test;
+	@JsonProperty("othersize")
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/model/Product.java]: why this ?

File: src/main/java/com/example/demo/service/ProductServiceimpl.java [hunk 11]
```diff
@@ -1,81 +1,77 @@
...
 
-	@Override
-	public Optional<Product> findEmployeeById(int id) {
-		// TODO Auto-generated method stub
-		return  product.findById(id);
 	}
 
 	@Override
-	public void createProduct(List<Product> emp) {
+	public Optional<Product> findEmployeeById(int id) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/service/ProductServiceimpl.java]: EmployeeBy ID ?? Where did employee come from ?

File: src/main/java/com/example/demo/service/ProductServiceimpl.java [hunk 12]
```diff
@@ -1,81 +1,77 @@
...
-	
-	@Autowired
-	Customerdao customer;
-	
-	public void createProduct(Product product2 , Customer customer1) {
-		// TODO Auto-generated method stub	
-	    customer.save(customer1);
-		product.save(product2);
-		
+	Productdao productdao;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/service/ProductServiceimpl.java]: Class name should be PascalCase and variable name should be camelCase

File: src/main/java/com/example/demo/service/ProductServiceimpl.java [hunk 13]
```diff
@@ -1,81 +1,77 @@
...
+	@Value("${spring.application.name}")
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/service/ProductServiceimpl.java]: where are you using name ?

File: src/main/java/com/example/demo/service/ProductServiceimpl.java [hunk 14]
```diff
@@ -1,81 +1,77 @@
...
+	private String name;
+
+	@Value("${spring.application.id}")
+	private String propId;
+
+	public void createProduct(Product product) {
+		// TODO Auto-generated method stub
+		Integer id;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/example/demo/service/ProductServiceimpl.java]: why does declaration and initialization needs to be different. Only separate them if they need to be separated

Do this instead

```
log.info("We are adding the properties id and user input id");
Integer id  = product.getId() + Integer.parseInt(getPropId());
```

**Review by AakashIsTheLimit — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/controller/WebController.java (hunk 1 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/controller/WebController.java]: Done.

File: src/main/java/com/example/demo/controller/WebController.java (hunk 2 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/controller/WebController.java]: Done.

File: src/main/java/com/example/demo/controller/WebController.java (hunk 3 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/controller/WebController.java]: Done.

File: src/main/java/com/example/demo/controller/WebController.java (hunk 4 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/controller/WebController.java]: Done.

File: src/main/java/com/example/demo/controller/WebController.java (hunk 5 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/controller/WebController.java]: Done.

File: src/main/java/com/example/demo/model/Product.java (hunk 10 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/model/Product.java]: Done.

File: src/main/java/com/example/demo/service/ProductServiceimpl.java (hunk 11 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/service/ProductServiceimpl.java]: Removed.

File: src/main/java/com/example/demo/service/ProductServiceimpl.java (hunk 12 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/service/ProductServiceimpl.java]: Done.

File: src/main/java/com/example/demo/service/ProductServiceimpl.java (hunk 13 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/service/ProductServiceimpl.java]: Removed.

File: src/main/java/com/example/demo/service/ProductServiceimpl.java (hunk 14 above)
Comment (KajalMathur) [reply src/main/java/com/example/demo/service/ProductServiceimpl.java]: Done.

File: src/main/java/com/example/demo/controller/WebController.java (hunk 1 above)
Comment (AakashIsTheLimit) [reply src/main/java/com/example/demo/controller/WebController.java]: Why dont you "Reply" to the comment.... you should reply to my comment and say "Done"

Like I replied to your done comment.

**Pull request closed without merging**